cmd/utils/nodecmd/util.go

```diff
@@ -0,0 +1,274 @@
+// Modifications Copyright 2023 The klaytn Authors
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package nodecmd
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/klaytn/klaytn/accounts/keystore"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/cmd/utils"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/crypto/sha3"
+	"github.com/klaytn/klaytn/governance"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/rlp"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const (
+	DECODE_EXTRA = "decode-extra"
+	DECODE_VOTE  = "decode-vote"
+	DECODE_GOV   = "decode-gov"
+	DECRYPT_KEY  = "decrypt-keystore"
+)
+
+var pgmInput = fmt.Sprintf(`
+  %17s  <header file (json format)>
+  %17s  <bytes>
+  %17s  <bytes>
+  %17s  <keystore path> <password>
+`, DECODE_EXTRA, DECODE_VOTE, DECODE_GOV, DECRYPT_KEY)
+
+var ErrInvalidParam = errors.New("Invalid length of parameter")
+
+var UtilCommand = cli.Command{
+	Action:    utils.MigrateFlags(decode),
+	Name:      "util",
+	Usage:     "offline utility" + pgmInput,
+	ArgsUsage: " ",
+	Category:  "MISCELLANEOUS COMMANDS",
+}
+
+func hex2Bytes(s string) []byte {
+	if data, err := hexutil.Decode(s); err == nil {
+		return data
+	} else {
+		panic(err)
+	}
+}
+
+func printUsage() error {
+	fmt.Printf("Usage: %s util <command> <args>", os.Args[0])
+	fmt.Println(pgmInput)
+	return ErrInvalidParam
+}
+
+func validateInput(ctx *cli.Context, decodeTyp string) error {
+	switch decodeTyp {
+	case DECODE_EXTRA, DECODE_VOTE, DECODE_GOV:
+		if len(ctx.Args()) != 2 {
+			return printUsage()
+		}
+	case "key":
+		if len(ctx.Args()) != 3 {
+			return printUsage()
+		}
+	}
+	return nil
+}
+
+func decode(ctx *cli.Context) error {
```

Comment:
You have added some `utility functions` for decode, right?
If I understand right, you can rename this command as `decode` and use these subcommands pattern.
Reference https://github.com/klaytn/klaytn/blob/dev/cmd/utils/nodecmd/accountcmd.go#L60

Reply:
Smarter way :) Refactored. Thanks.

Reply:
You can just put the `actual function name` to the `action field` of the subcommand. You don't need to switch each cases.